torch/_inductor/codegen/common.py

```diff
@@ -731,13 +742,16 @@ def __init__(self, args=None):
         self.stores = IndentedBuffer()
         self.cse = CSE(self.newvar_prefix, self.suffix)
         self.must_keep_buffers = set()
-        self.current_node = None
         self.store_buffer_names = set()
+        # set in set_current_node
+        self.current_node = None
+        self.bounds = None
 
     @contextlib.contextmanager
     def set_current_node(self, node):
         prior = self.current_node
         self.current_node = node
+        self.bounds = node._body.bounds().get_bounds()
```

Comment:
just curious, why set bounds here and not on initialization ?

Reply:
Because this is run on LoopBody, that is, on each node within a SchedulerNode. This was already the case in your initial implementation.